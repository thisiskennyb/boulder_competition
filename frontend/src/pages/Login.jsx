import { useState } from "react"
import { useNavigate } from "react-router-dom"
import axios from 'axios'
import { toast } from "react-toastify";
export default function Login() {
  const navigate = useNavigate()
  const [formData, setFormData ] = useState({
    "email": "",
    "password": "",
  })

  const {email, password} = formData

  const onChange = (event) => {
    setFormData({...formData, [event.target.name]: event.target.value})
  }

  const onSubmit = async (event) => {
    event.preventDefault();

    try {
      const response = await axios.post('http://localhost/api/jwt/create/', formData)
      console.log(response.data, 'Our response')
      document.cookie = `refresh=${response.data.refresh}`
      document.cookie = `access=${response.data.access}`
      toast.success('Welcome!')
      navigate('/profile')
      
    }
    catch (error) {
      console.log(error)
    }

  }
    return(
        <div className="flex min-h-full flex-1 flex-col justify-center px-6 py-12 lg:px-8">
        <div className="sm:mx-auto sm:w-full sm:max-w-sm">
          <img
            className="mx-auto h-10 w-auto"
            src="https://tailwindui.com/img/logos/mark.svg?color=indigo&shade=600"
            alt="Your Company"
          />
          <h2 className="mt-10 text-center text-2xl font-bold leading-9 tracking-tight text-gray-900">
            Sign in to your account
          </h2>
        </div>

        <div className="mt-10 sm:mx-auto sm:w-full sm:max-w-sm">
          <form className="space-y-6" onSubmit={onSubmit}>
            <div>
              <label htmlFor="email" className="block text-sm font-medium leading-6 text-gray-900">
                Email address
              </label>
              <div className="mt-2">
                <input
                  id="email"
                  className="block w-full rounded-md border-0 py-1.5 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6"
                  value={email}
                  onChange={onChange}
                  name="email"
                  type="email"
                  autoComplete="email"
                  required
                />
              </div>
            </div>

            <div>
              <div className="flex items-center justify-between">
                <label htmlFor="password" className="block text-sm font-medium leading-6 text-gray-900">
                  Password
                </label>
                <div className="text-sm">
                  <a href="#" className="font-semibold text-indigo-600 hover:text-indigo-500">
                    Forgot password?
                  </a>
                </div>
              </div>
              <div className="mt-2">
                <input
                  id="password"
                  className="block w-full rounded-md border-0 py-1.5 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6"
                  value={password}
                  onChange={onChange}
                  name="password"
                  type="password"
                  autoComplete="current-password"
                  required
                />
              </div>
            </div>

            <div>
              <button
                type="submit"
                className="flex w-full justify-center rounded-md bg-indigo-600 px-3 py-1.5 text-sm font-semibold leading-6 text-white shadow-sm hover:bg-indigo-500 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-indigo-600"
              >
                Sign in
              </button>
            </div>
          </form>

        </div>
      </div>
    )
}