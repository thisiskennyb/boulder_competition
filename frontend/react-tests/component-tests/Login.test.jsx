import {expect, test} from "vitest"
import Login from "../../src/pages/Login"

test("Login renders without crashing", ()=> {
    const component = <Login/>;
    expect(component).toMatchSnapshot()
})