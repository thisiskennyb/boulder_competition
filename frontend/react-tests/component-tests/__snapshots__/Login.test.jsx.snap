// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Login renders without crashing 1`] = `<Login />`;
